### ISSUE TITLE
Macro.to_string output does not match formatter output

### ISSUE THREAD

**Issue by thecodeboss** [elixir-lang/elixir#6984]
### Environment

* Elixir & Erlang versions (elixir --version):
```
Erlang/OTP 20 [erts-9.1.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.6.0-dev (794928e) (compiled with OTP 20)
```
* Operating system: 
```
MacOS High Sierra 10.13
```

### Current behavior

Running the following code:
```elixir
"""
defmodule Hello do
  def test() do
    1
  end
end
"""
|> Code.string_to_quoted!()
|> Macro.to_string()
```
The output looks like:
```elixir
defmodule(Hello) do
  def(test()) do
    1
  end
end
```

### Expected behavior

The output should match the format that the Elixir code formatter uses. In this case:
```elixir
defmodule Hello do
  def test() do
    1
  end
end
```
This is definitely a "nice to have" thing, obviously not urgent or required. I couldn't find any other issues related to this one, nor any discussions about whether it has already been considered. My apologies if this is a non-goal.

**Comment by josevalim** [comment]
Hi @thecodeboss!

They can't match because the formatter is based on a string, which gives us a lot more context, and the macro one relies on the AST. furthermore, Macro.to_string was designed to allow the result to be processed as we convert it to string, which would be tricky in the formatter.

TL;DR - different tools, different outputs.

have a good weekend!

**Comment by josevalim** [comment]
To further clarify, we are not against improving the output of Macro.to_string, you just shouldn't expect them ever to be the same.
